plugins/colorbutton/plugin.js

```diff
@@ -331,31 +359,47 @@ CKEDITOR.plugins.add( 'colorbutton', {
 					colorLabel = colorName;
 				}
 
-				output.push( '<td>' +
-					'<a class="cke_colorbox" _cke_focus=1 hidefocus=true' +
-						' title="', colorLabel, '"' +
-						' draggable="false"' +
-						' ondragstart="return false;"' + // Draggable attribute is buggy on Firefox.
-						' onclick="CKEDITOR.tools.callFunction(', clickFn, ',\'', colorCode, '\',\'', type, '\'); return false;"' +
-						' href="javascript:void(\'', colorCode, '\')"' +
-						' data-value="' + colorCode + '"' +
-						' role="option" aria-posinset="', ( i + 2 ), '" aria-setsize="', total, '">' +
-						'<span class="cke_colorbox" style="background-color:#', colorCode, '"></span>' +
-					'</a>' +
-					'</td>' );
+				output.push( '<td>' + generateTileHtml( {
+					colorLabel: colorLabel,
+					clickFn: clickFn,
+					colorCode: colorCode,
+					type: type,
+					position: i + 2,
+					setSize: total
+				} ) + '</td>' );
 			}
 
-			// Render the "More Colors" button.
+			// Render the "Content Colors" section.
+			output.push( '</tr>' +
+				'<tr>' +
+					'<td colspan=' + colorsPerRow + '" align="center">' +
+						'<span class="cke_colorcontent_label" style="display:none">Content Colors</span>' +
+					'</td>' +
+				'</tr>' +
+				'<tr class="cke_colorcontent_row">' );
+
 			if ( moreColorsEnabled ) {
+				// Render the "Custom Colors" section.
+				output.push( '</tr>' +
+					'<tr>' +
+						'<td colspan=' + colorsPerRow + '" align="center">' +
+							'<span class="cke_colorcustom_label" style="display:none">Custom Colors</span>' +
+						'</td>' +
+					'</tr>' +
+					'<tr class="cke_colorcustom_row">' );
+
+				// Render the "More Colors" button.
```

Comment:
This function is already pretty huge, adding anything on top of it only reduces readability. Could you extract template functions to clean up it a bit? Also, note [correct level of abstraction](http://principles-wiki.net/principles:single_level_of_abstraction) here.